Parse/src/main/java/com/parse/ParseObjectSubclassingController.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+package com.parse;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.Map;
+
+/* package */ class ParseObjectSubclassingController {
+  private static class ObjectSubclassInfo {
```

Comment:
Could we re-add this abstraction when we need it? Otherwise, this abstraction would look super over-architected to anyone trying to wrap their head around it.